### ISSUE TITLE
Compiling folsomite error

### ISSUE THREAD

**Issue by amit-siddhu** [discoproject/disco#646]
Compiling /home/disco/master/deps/folsomite/src/folsomite_graphite_client.erl failed:
/home/disco/master/deps/folsomite/src/folsomite_graphite_client.erl:none: error in parse transform 'lager_transform': {function_clause,
                                             [{lager_transform,
                                               '-walk_ast/2-fun-0-',
                                               [{typed_record_field,
                                                 {record_field,21,
                                                  {atom,21,tcp_socket}},
                                                 {remote_type,21,
                                                  [{atom,21,inet},
                                                   {atom,21,socket},
                                                   []]}}],
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {lists,map,2,
                                               [{file,"lists.erl"},
                                                {line,1239}]},
                                              {lager_transform,walk_ast,2,
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {compile,
                                               '-foldl_transform/2-anonymous-2-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,964}]},
                                              {compile,foldl_transform,2,
                                               [{file,"compile.erl"},
                                                {line,966}]},
                                              {compile,
                                               '-internal_comp/4-anonymous-1-',
                                               2,
                                               [{file,"compile.erl"},
                                                {line,321}]},
                                              {compile,fold_comp,3,
                                               [{file,"compile.erl"},
                                                {line,347}]},
                                              {compile,internal_comp,4,
                                               [{file,"compile.erl"},
                                                {line,331}]}]}
ERROR: compile failed while processing /home/disco/master/deps/folsomite: rebar_abort

**Comment by ghost** [comment]
Which version of erlang and rebar?

**Comment by amit-siddhu** [comment]
Erlang - 19
Rebar - 2.2.0

I managed to compile against erlang 18.3 but after handling many deprecation issues.

**Comment by jackieduckie** [comment]
I'm running into the same issue on **Erlang - 20, Rebar 3.5.1**

**Comment by ghost** [comment]
@jackieduckie thanks for the feedback do you have the traceback?

**Comment by jackieduckie** [comment]
==> goldrush (get-deps)
==> lager (get-deps)
==> bear (get-deps)
==> meck (get-deps)
==> folsom (get-deps)
==> folsomite (get-deps)
==> plists (get-deps)
==> mochiweb (get-deps)
==> master (get-deps)
==> goldrush (compile)
==> lager (compile)
==> bear (compile)
==> meck (compile)
==> folsom (compile)
==> folsomite (compile)
Compiling /Users/RuiningDong/disco/master/deps/folsomite/src/folsomite_graphite_client.erl failed:
/Users/RuiningDong/disco/master/deps/folsomite/src/folsomite_graphite_client.erl:none: error in parse transform 'lager_transform': {function_clause,
                                             [{lager_transform,
                                               '-walk_ast/2-fun-0-',
                                               [{typed_record_field,
                                                 {record_field,21,
                                                  {atom,21,tcp_socket}},
                                                 {remote_type,21,
                                                  [{atom,21,inet},
                                                   {atom,21,socket},
                                                   []]}}],
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {lists,map,2,
                                               [{file,"lists.erl"},
                                                {line,1239}]},
                                              {lager_transform,walk_ast,2,
                                               [{file,
                                                 "src/lager_transform.erl"},
                                                {line,60}]},
                                              {compile,
                                               '-foldl_transform/3-anonymous-2-',
                                               3,
                                               [{file,"compile.erl"},
                                                {line,1038}]},
                                              {compile,foldl_transform,3,
                                               [{file,"compile.erl"},
                                                {line,1041}]},
                                              {compile,
                                               '-internal_comp/5-anonymous-1-',
                                               3,
                                               [{file,"compile.erl"},
                                                {line,342}]},
                                              {compile,fold_comp,4,
                                               [{file,"compile.erl"},
                                                {line,369}]},
                                              {compile,internal_comp,5,
                                               [{file,"compile.erl"},
                                                {line,353}]}]}
ERROR: compile failed while processing /Users/RuiningDong/disco/master/deps/folsomite: rebar_abort
make: *** [master] Error 1

**Comment by ghost** [comment]
@jackieduckie this looks like the same error. can you try the fix created here : https://github.com/amit-siddhu/disco/commit/4763ac81b26a7dd50d3aa88f6db3fb83b5cfad52

Also see : https://github.com/discoproject/disco/pull/647
I haven't seen what the differences are between erl19 and erl 20.

If you know what changes have occurred between 19 and 20 feel free to open a pull request.

**Comment by jackieduckie** [comment]
Yes that solves part of the problem, combined with replacing now() with timestamp(), and updating the .config file of rebar, it works now. Thank you!
